### PR TITLE
Changes to events and auth

### PR DESCRIPTION
**Pull request by castaneda-ed**
Its showing as if I did changes to the auth.js file cause I did add tot he responses the user.id but then reverted. so everything is the same there, Where I did needed changes was in the events file, Added logic to delete users id if not going to the event and also add the user if going. Included also the attendingUsers in the GET events route.

### CODE REVIEW

File: api/auth.js [hunk 1]
```diff
@@ -8,116 +8,116 @@ const JWT_SECRET = process.env.JWT_SECRET;
```
Comment (castaneda-ed) [inline api/auth.js:8]: I did changes here but then reverted them cause bertha reminded me of the easier way. So everything is the same here.

File: api/events.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ const { authenticate } = require("./auth");
 //get all events
 router.get("/", async (req, res, next) => {
   try {
-    const events = await prisma.event.findMany();
```
Comment (castaneda-ed) [inline api/events.js:11]: Here included the attendingUsers to the request.

File: api/events.js [hunk 3]
```diff
@@ -30,22 +32,28 @@ router.get("/:id/", async (req, res, next) => {
...
     const event = await prisma.event.update({
       where: {
         id: +id,
       },
       data: {
-        attendingUsers: {
-          connect: {
-            id: req.user.id,
-          },
-        },
```
Comment (castaneda-ed) [inline api/events.js:48]: logic for connecting the users id if attending is true and disconnecting if attending is false.

File: api/auth.js [hunk 4]
```diff
@@ -8,116 +8,116 @@ const JWT_SECRET = process.env.JWT_SECRET;
...
-    return jwt.sign({ id }, JWT_SECRET, { expiresIn: "1d" });
+  return jwt.sign({ id }, JWT_SECRET, { expiresIn: "1d" });
 }
 
 // Email validation regex
 const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
 
 // token checking middleware
 router.use(async (req, res, next) => {
-    const authHeader = req.headers.authorization;
```
Comment (castaneda-ed) [inline api/auth.js:19]: I did changes here but then reverted them cause bertha reminded me of the easier way. So everything is the same here.

File: api/events.js [hunk 5]
```diff
@@ -8,7 +8,9 @@ const { authenticate } = require("./auth");
 //get all events
 router.get("/", async (req, res, next) => {
   try {
-    const events = await prisma.event.findMany();
+    const events = await prisma.event.findMany({
+      include: { attendingUsers: true },
+    });
     res.json(events);
   } catch (e) {
     next(e);
```
Comment (castaneda-ed) [inline api/events.js:16]: Here included the attendingUsers to the request.

File: api/events.js [hunk 6]
```diff
@@ -30,22 +32,28 @@ router.get("/:id/", async (req, res, next) => {
...
+              },
+            }
+          : {
+              disconnect: {
+                id: req.user.id,
+              },
+            },
       },
     });
     res.status(201).json(event);
```
Comment (castaneda-ed) [inline api/events.js:59]: logic for connecting the users id if attending is true and disconnecting if attending is false.

**Review by beefchief-code — APPROVED** [review]
looks good to me

**Review by wangbertha — APPROVED** [review]

**Pull request merged**